src/gmail.js

```diff
@@ -1277,6 +1277,24 @@ var Gmail = function(localJQuery) {
         $(window.document).bind('DOMNodeInserted', function(e) {
           api.tools.insertion_observer(e.target, api.tracker.dom_observers, api.tracker.dom_observer_map);
         });
+
+        // recipient_change also needs to listen to removals
+        var mutationObserver = new MutationObserver(function(mutations) {
+          for (var i = 0; i < mutations.length; i++) {
+            var mutation = mutations[i];
+            var removedNodes = mutation.removedNodes;
+            for (var j = 0; j < removedNodes.length; j++) {
+              var removedNode = removedNodes[j];
+              if (removedNode.className == 'vR') {
+                var observer = api.tracker.dom_observer_map['vR'];
+                var handler = api.tracker.dom_observers.recipient_change.handler;
+                api.observe.trigger_dom(observer, $(mutation.target), handler);
+              }
+            }
+          }
+        });
+        mutationObserver.observe($('div.dw')[0], {subtree: true, childList: true});
```

Comment:
The issue is that I'm creating the observer when the script loads and before compose windows are opened, so it needs to be on something high level. To add it to compose windows, I'd need to add a new observer to each compose window when it is opened, which seems a little weird / cumbersome. I've also noticed the script doesn't always catch compose open events, if I click the compose button very quickly after gmail loads, or if I refresh a page with a compose window already open.